src/core/expr/cumsum.cc

```diff
@@ -0,0 +1,120 @@
+#include "expr/fexpr_func.h"
+#include "column/virtual.h"
+#include "expr/eval_context.h"
+#include "expr/workframe.h"
+#include "stype.h"
+#include "python/xargs.h"
+#include "documentation.h"
+#include <numeric>
+#include <vector>
+
+namespace dt {
+namespace expr {
+
+template <typename T>
+class Column_cumsum : public Virtual_ColumnImpl{
+    private:
+      Column acol_;
+
+    public:
+      Column_cumsum(Column&& a)
+        : Virtual_ColumnImpl(a.nrows(), a.stype()),
+          acol_(std::move(a))
+
+    {
+      xassert(acol_.can_be_read_as<T>());
+    }
+
+      ColumnImpl* clone() const override {
+          return new Column_cumsum(Column(acol_));
+      }
+
+      size_t n_children() const noexcept override {
+      return 1;
+    }
+
+    const Column& child(size_t) const override {
+      return acol_;
+    }
+
+
+    bool get_element(size_t i, T* out) const override {
+      T a;
+      bool isvalid = acol_.get_element(i, &a);
+      if (isvalid) {
+        *out = static_cast<T>(acol_.nrows()); // testing a dummy output here to return no of rows
+      }
+      return isvalid;
+    }
+
+};
+
+
+
+
+
+class FExpr_cumsum : public FExpr_Func {
+  private:
+    ptrExpr a_;
+
+  public:
+    FExpr_cumsum(ptrExpr&& a)
+      : a_(std::move(a)) {}
+
+    std::string repr() const override{
+        std::string out = "cumsum(";
+        out += a_->repr();
+        out += ')';
+        return out;
+    };
+    Workframe evaluate_n(EvalContext& ctx) const override {
+        Workframe awf = a_->evaluate_n(ctx);
+
+        auto gmode = awf.get_grouping_mode();
+        Workframe outputs(ctx);
+        for (size_t i = 0; i < awf.ncols(); ++i) {
+            Column rescol = evaluate1(awf.retrieve_column(i));
+            outputs.add_column(std::move(rescol), std::string(), gmode);
+         }
+
+        return outputs;
+    };
+
+    Column evaluate1(Column&& a) const {
+
+  switch (a.stype()) {
+    case SType::BOOL:
+    case SType::INT8:
+    case SType::INT16:
+    case SType::INT32: return a;
+    case SType::INT64: return a;
+    case SType::FLOAT32: return a;
+    case SType::FLOAT64: return a;
```

Comment:
The issue here is that you always return `a`, that is the original column. What you should do is to call `make(a)` and then return the result.

Reply:
@oleksiyskononenko  is this what you mean :``case SType::INT32: make(a);``

Reply:
You should also return it, so something like this will work `case SType::INT32: return make(a);`

Reply:
thanks @oleksiyskononenko ; my next step is how to pass a vector that is the same size ; so instead of return a.nrows(), I wish to return [2, 4, 5, 6] or some other vector that is the same size. I'll gladly take any pointers/guide

Reply:
Take a look at documentation "[Creating a new FExpr](https://datatable.readthedocs.io/en/latest/develop/create-fexpr.html)". Though for `cumsum()` it could be more efficient to fill out all the column values with one pass. Otherwise, you will have to re-calculate partial sums over and over again.

Reply:
@oleksiyskononenko still not getting the flow, so some pointers/guidance is definitely needed for me. I thought I could use `partial_sum` func here, but not sure how to do that. The other thought would be to get the first value and store in a variable, and then cumulatively add the rest of the values in the column; I do not know how to select that first value

Reply:
`partial_sum` would require a vector or some similar data structure. However, we don't store data in vectors and to access any given element one should use `get_element()` method. Take a look at the link I posted above for an example of how `get_element()` should be used.

Reply:
@oleksiyskononenko made some edits; my outputs are all 0 though. Kindly have a look, when you can and guide me on how to fix it. thanks